client/src/app.tsx

```diff
@@ -119,6 +101,18 @@ export function App(appProps) {
     document.documentElement.setAttribute("lang", locale);
   }, [appProps.locale, localeMatch]);
 
+  const [pageNotFound, setPageNotFound] = React.useState<boolean>(
+    appProps.pageNotFound
+  );
+  const { pathname } = useLocation();
+  const initialPathname = React.useRef(pathname);
+
+  React.useEffect(() => {
+    if (initialPathname.current !== pathname) {
+      setPageNotFound(false);
+    }
```

Comment:
@danielhjacobs This change might fix 5. of the referenced issue?

```suggestion
    setPageNotFound(initialPathname.current === pathname);
```

Reply:
See: https://github.com/mdn/yari/pull/6459